src/Types.jl

```diff
@@ -109,11 +110,12 @@ function PackageSpec(; name::Union{Nothing,AbstractString} = nothing,
                        repo::GitRepo = GitRepo(),
                        path::Union{Nothing,AbstractString} = nothing,
                        pinned::Bool = false,
+                       yanked::Bool = false,
```

Comment:
I am not a big fan of adding this to the PackageSpec. 

Yanking is so rare and having a yanked version in the manifest is even rarer etc it feels kind of overkill.